src/uct/ib/ud/accel/ud_mlx5.c

```diff
@@ -285,7 +287,15 @@ uct_ud_mlx5_ep_inline_iov_post(uct_ep_h tl_ep, uint8_t am_id,
     memcpy(neth + 1, header, header_size);
 
     /* copy inline "data" */
-    uct_ib_mlx5_inline_copy(wqe_data, data, data_size, &iface->tx.wq);
+    if (data_size > 0) {
+        uct_ib_mlx5_inline_copy(wqe_data, data, data_size, &iface->tx.wq);
+    }
+
+    /* copy inline iov */
+    if (inl_iovcnt > 0) {
+        uct_ib_mlx5_inline_iov_copy(wqe_data, inl_iov, inl_iovcnt, inl_iov_size,
+                                    &iface->tx.wq);
+    }
```

Comment:
would need to shift `wqe_data` before copying inline iov, or make sure that both `data` and `inl_iov` can't be non NULL (by adding assert for instance)